src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertificate.swift

```diff
@@ -280,6 +284,19 @@ final class HealthCertificate: Codable, Equatable, Comparable, RecycleBinIdentif
 			return hashData == $0.hash
 		}
 	}
+	
+	func belongsToSamePerson(_ other: HealthCertificate) -> Bool {
+		let hasSameDateOfBirth = self.trimmedDateOfBirth == other.trimmedDateOfBirth
+		
+		let givenNameCompontents = Set<String>(self.name.givenNameGroupingComponents ?? [])
+		let otherGivenNameCompontents = other.name.givenNameGroupingComponents ?? []
```

Comment:
We could move the fallback to `[]` into `givenNameGroupingComponents`

Reply:
done